From ea85022f763a7a9cda2196f241db8a4319a46f55 Mon Sep 17 00:00:00 2001
From: AF9 <af9@localhost>
Date: Tue, 27 Oct 2026 19:18:54 +0000
Subject: [PATCH] Remove the Distillation Tower: AF9's Distillation Facility
 runs its recipes (the Large Distillery is built from an IV Distillery now)
 (7.5.3-af9.27)

---
 gradle.properties                             |  2 +-
 .../common/data/machines/GTMultiMachines.java | 71 -------------------
 .../gtceu/data/recipe/misc/GCYMRecipes.java   |  2 +-
 .../recipe/misc/MetaTileEntityLoader.java     |  5 --
 4 files changed, 2 insertions(+), 78 deletions(-)

diff --git a/gradle.properties b/gradle.properties
index 38fa03d..91f48f2 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -6,7 +6,7 @@ org.gradle.caching = true
 # Mod Info
 mod_id = gtceu
 mod_name = GregTech
-mod_version = 7.5.3-af9.26
+mod_version = 7.5.3-af9.27
 mod_description = GregTech CE Unofficial, ported from 1.12.2
 mod_license = LGPL-3.0 license
 mod_url = https://github.com/GregTechCEu/GregTech-Modern/
diff --git a/src/main/java/com/gregtechceu/gtceu/common/data/machines/GTMultiMachines.java b/src/main/java/com/gregtechceu/gtceu/common/data/machines/GTMultiMachines.java
index 9456d1c..7033092 100644
--- a/src/main/java/com/gregtechceu/gtceu/common/data/machines/GTMultiMachines.java
+++ b/src/main/java/com/gregtechceu/gtceu/common/data/machines/GTMultiMachines.java
@@ -443,77 +443,6 @@ public class GTMultiMachines {
             })
             .register();
 
-    public static final MultiblockMachineDefinition DISTILLATION_TOWER = REGISTRATE
-            .multiblock("distillation_tower", DistillationTowerMachine::new)
-            .rotationState(RotationState.NON_Y_AXIS)
-            .recipeType(GTRecipeTypes.DISTILLATION_RECIPES)
-            .recipeModifiers(OC_NON_PERFECT_SUBTICK, BATCH_MODE)
-            .appearanceBlock(CASING_STAINLESS_CLEAN)
-            .pattern(definition -> {
-                TraceabilityPredicate exportPredicate = abilities(PartAbility.EXPORT_FLUIDS_1X);
-                if (GTCEu.Mods.isAE2Loaded()) {
-                    exportPredicate = exportPredicate.or(blocks(GTAEMachines.FLUID_EXPORT_HATCH_ME.get()));
-                }
-                exportPredicate.setMaxLayerLimited(1);
-                TraceabilityPredicate maint = autoAbilities(true, false, false)
-                        .setMaxGlobalLimited(1);
-                return FactoryBlockPattern.start(RIGHT, BACK, UP)
-                        .aisle("YSY", "YYY", "YYY")
-                        .aisle("ZZZ", "Z#Z", "ZZZ")
-                        .aisle("XXX", "X#X", "XXX").setRepeatable(0, 10)
-                        .aisle("XXX", "XXX", "XXX")
-                        .where('S', Predicates.controller(blocks(definition.getBlock())))
-                        .where('Y', blocks(CASING_STAINLESS_CLEAN.get())
-                                .or(Predicates.abilities(PartAbility.EXPORT_ITEMS).setMaxGlobalLimited(1))
-                                .or(Predicates.abilities(PartAbility.INPUT_ENERGY).setMinGlobalLimited(1)
-                                        .setMaxGlobalLimited(2))
-                                .or(Predicates.abilities(PartAbility.IMPORT_FLUIDS).setExactLimit(1))
-                                .or(maint))
-                        .where('Z', blocks(CASING_STAINLESS_CLEAN.get())
-                                .or(exportPredicate)
-                                .or(maint))
-                        .where('X', blocks(CASING_STAINLESS_CLEAN.get()).or(exportPredicate))
-                        .where('#', Predicates.air())
-                        .build();
-            })
-            .shapeInfos(definition -> {
-                List<MultiblockShapeInfo> shapeInfos = new ArrayList<>();
-                var builder = MultiblockShapeInfo.builder()
-                        .where('C', definition, Direction.NORTH)
-                        .where('S', CASING_STAINLESS_CLEAN.getDefaultState())
-                        .where('X', ITEM_EXPORT_BUS[HV], Direction.NORTH)
-                        .where('I', FLUID_IMPORT_HATCH[HV], Direction.NORTH)
-                        .where('E', ENERGY_INPUT_HATCH[HV], Direction.SOUTH)
-                        .where('M', MAINTENANCE_HATCH, Direction.SOUTH)
-                        .where('#', Blocks.AIR.defaultBlockState())
-                        .where('F', FLUID_EXPORT_HATCH[HV], Direction.SOUTH);
-                List<String> front = new ArrayList<>(15);
-                front.add("XCI");
-                front.add("SSS");
-                List<String> middle = new ArrayList<>(15);
-                middle.add("SSS");
-                middle.add("SSS");
-                List<String> back = new ArrayList<>(15);
-                back.add("MES");
-                back.add("SFS");
-                for (int i = 1; i <= 11; ++i) {
-                    front.add("SSS");
-                    middle.add(1, "S#S");
-                    back.add("SFS");
-                    var copy = builder.shallowCopy()
-                            .aisle(front.toArray(String[]::new))
-                            .aisle(middle.toArray(String[]::new))
-                            .aisle(back.toArray(String[]::new));
-                    shapeInfos.add(copy.build());
-                }
-                return shapeInfos;
-            })
-            .allowExtendedFacing(false)
-            .partSorter(Comparator.comparingInt(p -> p.self().getPos().getY()))
-            .workableCasingModel(GTCEu.id("block/casings/solid/machine_casing_clean_stainless_steel"),
-                    GTCEu.id("block/multiblock/distillation_tower"))
-            .register();
-
     public static final MultiblockMachineDefinition VACUUM_FREEZER = REGISTRATE
             .multiblock("vacuum_freezer", WorkableElectricMultiblockMachine::new)
             .rotationState(RotationState.ALL)
diff --git a/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/GCYMRecipes.java b/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/GCYMRecipes.java
index e01d000..f640e86 100644
--- a/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/GCYMRecipes.java
+++ b/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/GCYMRecipes.java
@@ -129,7 +129,7 @@ public class GCYMRecipes {
                 FERMENTER[IV].asStack(), 'E', ELECTRIC_PUMP_IV.asStack(), 'B', BREWERY[IV].asStack(), 'H',
                 FLUID_HEATER[IV].asStack(), 'K', new MaterialEntry(cableGtSingle, Platinum));
         VanillaRecipeHelper.addShapedRecipe(provider, true, "large_distillery", LARGE_DISTILLERY.asStack(), "PZP",
-                "EDE", "PZP", 'Z', CustomTags.IV_CIRCUITS, 'D', GTMultiMachines.DISTILLATION_TOWER.asStack(), 'E',
+                "EDE", "PZP", 'Z', CustomTags.IV_CIRCUITS, 'D', DISTILLERY[IV].asStack(), 'E',
                 ELECTRIC_PUMP_IV.asStack(), 'P', ChemicalHelper.get(pipeLargeFluid, Iridium));
         VanillaRecipeHelper.addShapedRecipe(provider, true, "large_extractor", LARGE_EXTRACTOR.asStack(), "PTP", "EZC",
                 "BKB", 'Z', CustomTags.IV_CIRCUITS, 'B', ELECTRIC_PISTON_IV.asStack(), 'P', ELECTRIC_PUMP_IV.asStack(),
diff --git a/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/MetaTileEntityLoader.java b/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/MetaTileEntityLoader.java
index 2825e6d..2f5f252 100644
--- a/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/MetaTileEntityLoader.java
+++ b/src/main/java/com/gregtechceu/gtceu/data/recipe/misc/MetaTileEntityLoader.java
@@ -589,11 +589,6 @@ public class MetaTileEntityLoader {
                 GTMultiMachines.IMPLOSION_COMPRESSOR.asStack(), "OOO", "CMC", "WCW", 'M',
                 GTBlocks.CASING_STEEL_SOLID.asStack(), 'O', new MaterialEntry(TagPrefix.rock, GTMaterials.Obsidian),
                 'C', CustomTags.HV_CIRCUITS, 'W', new MaterialEntry(TagPrefix.cableGtSingle, GTMaterials.Gold));
-        VanillaRecipeHelper.addShapedRecipe(provider, true, "distillation_tower",
-                GTMultiMachines.DISTILLATION_TOWER.asStack(), "CBC", "FMF", "CBC", 'M', GTMachines.HULL[HV].asStack(),
-                'B',
-                new MaterialEntry(TagPrefix.pipeLargeFluid, GTMaterials.StainlessSteel), 'C', CustomTags.EV_CIRCUITS,
-                'F', GTItems.ELECTRIC_PUMP_HV);
         VanillaRecipeHelper.addShapedRecipe(provider, true, "cracking_unit", GTMultiMachines.CRACKER.asStack(), "CEC",
                 "PHP",
                 "CEC", 'C', GTBlocks.COIL_CUPRONICKEL.asStack(), 'E', GTItems.ELECTRIC_PUMP_HV.asStack(), 'P',
-- 
2.43.0

